extension/chrome/settings/modules/contacts.ts

```diff
@@ -109,26 +110,44 @@ Catch.try(async () => {
 
   const actionProcessBulkImportTextInput = async () => {
     try {
-      const value = String($('#bulk_import .input_pubkey').val());
+      const value = Str.normalize(String($('#bulk_import .input_pubkey').val())).trim();
+      if (!value) {
+        await Ui.modal.warning('Please paste public key(s).');
+        return;
+      }
       const normalizedLongid = KeyImportUi.normalizeLongId(value);
-      let pubkey: string;
+      let pub: string;
       if (normalizedLongid) {
         const data = await Keyserver.lookupLongid(acctEmail, normalizedLongid);
         if (data.pubkey) {
-          pubkey = data.pubkey;
+          pub = data.pubkey;
         } else {
           await Ui.modal.warning('Could not find any Public Key in our public records that matches this fingerprint or longid');
           return;
         }
       } else {
-        pubkey = value;
+        pub = value;
       }
-      const replacedHtmlSafe = XssSafeFactory.replaceRenderableMsgBlocks(factory, pubkey);
-      if (replacedHtmlSafe && replacedHtmlSafe !== value) {
-        $('#bulk_import #processed').html(replacedHtmlSafe).css('display', 'block'); // xss-safe-factory
+      let { blocks } = Pgp.armor.detectBlocks(pub);
+      blocks = blocks.filter((b, i) => blocks.findIndex(f => f.content === b.content) === i); // remove duplicates
```

Comment:
I didn't think of removing duplicates. That's a good idea!

Reply:
Thanks